app/models/cinstance.rb

```diff
@@ -140,7 +140,8 @@ def validate_plan_is_unique?
   }
 
   def self.provided_by(account)
-    joins(:service).references(:service).merge(Service.of_account(account)).readonly(false)
+    non_buyer_scope = account.master? ? where("user_account_id NOT IN (?)", account.users.pluck(:id)) : all
+    non_buyer_scope.joins(:service).references(:service).merge(Service.of_account(account)).readonly(false)
```

Comment:
I'd say create do not change this method. But instead where it is used do the change, depending if it is master or not.